types/node/vm.d.ts

```diff
@@ -69,7 +69,7 @@ declare module "vm" {
          * [Support of dynamic `import()` in compilation APIs](https://nodejs.org/docs/latest-v22.x/api/vm.html#support-of-dynamic-import-in-compilation-apis).
          */
         importModuleDynamically?:
-            | ((specifier: string, script: Script, importAttributes: ImportAttributes) => Module)
+            | ((specifier: string, script: Script, importAttributes: ImportAttributes) => Promise<Module>)
```

Comment:
My reading of the way this is wrapped is that the provided function isn't _obliged_ to be asynchronous, it's just [treated as an awaitable](https://github.com/nodejs/node/blob/a7f648c8ba77b542dff20e717ab3e005425fa1a6/lib/internal/vm/module.js#L436).

```suggestion
            | ((specifier: string, script: Script, importAttributes: ImportAttributes) => Module | Promise<Module>)
```

The internal Node.js tests have examples of both.